### ISSUE TITLE
getGameByTag() function 

### ISSUE THREAD

**Issue by codesungrape** [codesungrape/RESTful_Practice1#21]
This MVP1 function searches a array with nested objects for tag matches. It iterates through nested objects, comparing their 'tag' values against an input tag. When found, matching objects are collected into an array. Returns empty array if no matches.

I am getting a response of 'No games found- sorry!' indicating the catch() is being triggered.

**Comment by codesungrape** [comment]
The function sanitize(tags) logs expected value. 
However, sanitize(game.tags) seems to trigger the catch block and returns 'No games found- sorry!'. This means that game.tags, while it accesses the tag values within it, it doesn't loop through these values. 
I need to figure out a way to do this. 
I tried to use Object.values.

**Comment by codesungrape** [comment]
I was using Object.values but I needed to use object.keys() as the data I am comparing the 'tag' to is the 'key' part of the object.

**Comment by codesungrape** [comment]
The problem was with the way the sanitise() fucntion was working as well. Because Object.keys returns an array, but the sanitise function expects a string, it caused a TypeError.

- In order to correct this, I have included a type checking if stamens in the snaitize() which tells me if there is the wrong input and what type of input there is instead of the expected string input.
    if (typeof inputString !== "string") {
    console.error("Expected a string but got:", typeof inputString);
    return ""; // Return a default value if inputString is not a string
  }

- I have also chained a .map(sanitize), which calls the sanitize function on each string value as it accesses the returned array of Object.keys(game.tags). 
 cachedGameData.filter((game) => {
      if (Object.keys(game.tags).map(sanitize).includes(sanitize(tag))) {
        result.push(game);
      }
    });
